### PR TITLE
feat: support telegram

### LINKED ISSUE #170: Add support to send notifications to telegram

### PR DESCRIPTION
**Pull request by mangoGoForward**
fix #170

### CODE REVIEW

File: pkg/internal/telegram/types.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			Template: internal.Template{
+				TmplName: *telegram.Template,
+				TmplText: telegram.TmplText,
+			},
+		},
+		Channels:       telegram.Channels,
+		MentionedUsers: telegram.MentionedUsers,
+	}
+
+	return r
```
Comment (wanjunlei) [inline pkg/internal/telegram/types.go:45]: The receiver needs to set resourceversion too like [this](https://github.com/kubesphere/notification-manager/blob/master/pkg/internal/dingtalk/types.go#L55).

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: pkg/notify/notifier/telegram/telegram.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	"github.com/kubesphere/notification-manager/pkg/internal"
+	"github.com/kubesphere/notification-manager/pkg/internal/telegram"
+	"github.com/kubesphere/notification-manager/pkg/notify/notifier"
+	"github.com/kubesphere/notification-manager/pkg/template"
+	"github.com/kubesphere/notification-manager/pkg/utils"
+)
+
+const (
+	DefaultSendTimeout = time.Second * 3
+	URL                = "https://api.telegram.org/bot%s/sendMessage"
```
Comment (wanjunlei) [inline pkg/notify/notifier/telegram/telegram.go:23]: Does it miss a ":" between `bot` and `token`?

Comment (mangoGoForward) [reply pkg/notify/notifier/telegram/telegram.go:23]: the token is `123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11`

File: pkg/notify/notifier/dingtalk/throttle.go
Comment (Gentleelephant) [inline pkg/notify/notifier/dingtalk/throttle.go:1]: Are the changes here necessary? If not please remove it.

Comment (mangoGoForward) [reply pkg/notify/notifier/dingtalk/throttle.go:1]: done

File: sidecar/kubernetes/cmd/main.go [hunk 3]
```diff
@@ -104,13 +104,13 @@ func handler(req *restful.Request, resp *restful.Response) {
 	responseWithJson(resp, tenants)
 }
 
-//readiness
```
Comment (Gentleelephant) [inline sidecar/kubernetes/cmd/main.go:107]: Same problem.

Comment (mangoGoForward) [reply sidecar/kubernetes/cmd/main.go:107]: I think it's not a problem, it fixed by lint automatically and the between of doc commend and `//` should have a empty space

**Review by Gentleelephant — APPROVED** [review]
approve

**Review by wanjunlei — APPROVED** [review]

**Comment by wanjunlei** [comment]
@mangoGoForward Thank you for contributing to this project!

**Pull request merged**